Sources/MatrixClient/API/Room/RoomSendMessage.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  File.swift
+//
+//
+//  Created by Finn Behrens on 25.03.22.
+//
+
+import Foundation
+
+public struct MatrixRoomSendEventRequest {
+    public var body: MatrixEvent
+
+    public var roomID: String
+}
+
+extension MatrixRoomSendEventRequest: MatrixRequest {
+    public typealias Response = MatrixRoomSendEvent
+
+    public func components(for homeserver: MatrixHomeserver, with txID: Int) throws -> URLComponents {
+        homeserver.path("/_matrix/client/v3/rooms/\(roomID)/send/\(type(of: body).type)/\(txID)")
+    }
+
+    public static var httpMethod: HttpMethod {
+        .PUT
+    }
+
+    public static var requiresAuth: Bool {
+        true
+    }
+}
+
+extension MatrixRoomSendEventRequest: Codable {
+    public init(from _: Decoder) throws {
+        fatalError("todo")
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        try body.encode(to: encoder)
```

Comment:
If `body` is a `MatrixEvent` then isn't this going to encode as
```
"content": {
    /* event content goes here */
}
```

instead of what the request expects:

```
{
    /* event content goes here */
}
```

Reply:
I'm not sure that's true. Would you be okay with leaving it like this, until we have enough support in Nio to test it?